### PR TITLE
Defect/536 teacher target report subgroups

### PR DESCRIPTION
**Pull request by esotrope**
image

image

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/subgroup/subgroup.mapper.ts [hunk 1]
```diff
@@ -190,11 +190,24 @@ const DimensionConfigurations: DimensionConfiguration[] = [
     getTranslationCode: value => `common.ethnicity.${value}`
   },
   {
+    type: 'LimitedEnglishProficiency',
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/subgroup/subgroup.mapper.ts]: went with a superset for this as i want to eventually move away from the acronyms

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts [hunk 2]
```diff
@@ -44,20 +44,16 @@ export class TargetStatisticsCalculator {
     allTargets: Target[],
     targetScoreExams: TargetScoreExam[]
   ): AggregateTargetScoreRow[] {
-    // setup the placeholders to aggregate into
-    let groupedScores = this.generateOverallTargets(subjectCode, allTargets);
+    targetScoreExams = targetScoreExams || [];
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: misc code styles

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts [hunk 3]
```diff
@@ -193,27 +186,29 @@ export class TargetStatisticsCalculator {
...
+        case 'EnglishLanguageAcquisitionStatus':
+          // adding NULL since we have Not Stated data but not a filter option for it
+          subgroupValues = [...subgroupOptions.elasCodes, null];
           break;
-        case 'LEP':
+        case 'EconomicDisadvantage':
+        case 'LimitedEnglishProficiency':
         case 'Section504':
-        case 'IEP':
+        case 'IndividualEducationPlan':
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: real change in the file was to map these to the new sting values

File: webapp/src/main/webapp/src/assets/i18n/en.json [hunk 4]
```diff
@@ -424,11 +424,16 @@
       "EconomicDisadvantage": "Economic Disadvantage",
       "Ethnicity": "Ethnicity",
       "Gender": "Gender",
+      "IndividualEducationPlan": "IEP",
       "IEP": "IEP",
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/assets/i18n/en.json]: I know you don't let initialisms but this mapping change is ... amusing.  ;)

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts [hunk 5]
```diff
@@ -44,20 +44,16 @@ export class TargetStatisticsCalculator {
...
-      let index = groupedScores.findIndex(x => x.targetId == exam.targetId);
-      if (index !== -1) {
-        groupedScores[index].standardMetScores.push(
-          exam.standardMetRelativeResidualScore
-        );
-        groupedScores[index].studentScores.push(
-          exam.studentRelativeResidualScore
-        );
+      const scores = groupedScores.find(x => x.targetId == exam.targetId);
+      if (scores != null) {
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: `!==`?

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts [hunk 6]
```diff
@@ -254,27 +249,29 @@ export class TargetStatisticsCalculator {
...
+      case 'PrimaryLanguage':
         return exam.languageCode;
-      case 'MilitaryConnectedCode':
+      case 'MilitaryStudentIdentifier':
         return exam.militaryConnectedCode;
 
       // this is returned as undefined, but for subgrouping it needs to be null to match
-      case 'ELAS':
+      case 'EnglishLanguageAcquisitionStatus':
         return exam.elasCode! + null ? exam.elasCode : null;
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: Long case statements like these make me feel like it should be a dictionary or a map.

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts (hunk 5 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: `.find()` returns undefined when it fails i think.
`!= null` is "not null or undefined" in js

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts (hunk 6 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: i feel the same way. and if i wrote it there probably would be. But i'm not that ambitious today : J

File: webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts (hunk 5 above)
Comment (bgsavage) [reply webapp/src/main/webapp/src/app/assessments/results/target-statistics-calculator.ts]: Just making sure it wasn't an oversight, you were intentionally being implicit then :P

**Pull request merged**